src/main/java/com/powsybl/openloadflow/network/LfAction.java

```diff
@@ -319,16 +316,20 @@ private void applyTapPositionChange() {
     }
 
     private void applyLoadShift() {
-        LfLoad load = loadShift.load();
-        if (!load.isOriginalLoadDisabled(loadShift.loadId())) {
+        Load load = loadShift.load();
+        LfLoad lfLoad = loadShift.lfLoad();
+        if (!lfLoad.isOriginalLoadDisabled(load.getId())) {
             PowerShift shift = loadShift.powerShift();
-            load.setTargetP(load.getTargetP() + shift.getActive());
-            load.setTargetQ(load.getTargetQ() + shift.getReactive());
-            load.setAbsVariableTargetP(load.getAbsVariableTargetP()
-                    + Math.signum(shift.getActive()) * Math.abs(shift.getVariableActive()));
+            lfLoad.setTargetP(lfLoad.getTargetP() + shift.getActive());
+            lfLoad.setTargetQ(lfLoad.getTargetQ() + shift.getReactive());
+            lfLoad.setAbsVariableTargetP(lfLoad.getAbsVariableTargetP() + getUpdatedLoadAbsVariableTargetP(load, shift));
         }
     }
 
+    private static double getUpdatedLoadAbsVariableTargetP(Load load, PowerShift shift) {
+        return load.isFictitious() || LoadType.FICTITIOUS.equals(load.getLoadType()) ? 0.0 : Math.signum(shift.getActive()) * Math.abs(shift.getVariableActive());
+    }
```

Comment:
This implementation might be confusing to some users because it silently does something else than the user asked.
If the user created a LoadShift with a fictive load, why do you want not to honnor this ?

In case you really want to do this, it would bet better to generate and error when the LoadShift is created with a fictive load, or at least Warning in the log.

Also it the facto of ignoring fictive load a feature desired by ALL users of load compensation ? If not this  implementation that is not controlled by a parameter  might be considered as a breaking change.

Reply:
The shift is applied for any type of load.
```java
lfLoad.setTargetP(lfLoad.getTargetP() + shift.getActive());
lfLoad.setTargetQ(lfLoad.getTargetQ() + shift.getReactive());
lfLoad.setAbsVariableTargetP(lfLoad.getAbsVariableTargetP() + getUpdatedLoadAbsVariableTargetP(loadIsFictitious, shift));
```

The only difference is that the `absVariableTargetP` is not updated if the load is fictitious.
From what I understood, this `absVariableTargetP` is only used for slack distribution on loads.
Therefore, it is aligned with what I'm proposing.

Your point is still valid though: maybe I should also add a parameter to define if the user wants to distribute slack on fictitious loads or not.
I assumed that would always be the case, but maybe it is not...

What do you think about that?

Reply:
I think there are two issues. One for Actions, and one for compensation.
- For Actions, I would consider that if a user created an action on a fictive load, it is his decision and the software should just do it. 
If however there are good arguments to say that this can only be an error (I do not see them) the software should not silently ignore this. Choices are then throwing an Exception or printing a warning/error in the log. But not silently doing something else.

For compensation, they key is to know if some users  have valid use cases to compensate on fictive loads. I don't hace the answer. If nobody has a need, it's OK to make it a default BUT AT THE MINIMUM make the change visible in the relase notes  If valid use cases to compensate on fictive loads exist then you have, unfortunately,  to add a parameter to the 80 or more already existing... ANd to select a good default.

Reply:
Thanks for the explanation. Talking of several loads on a bus there are other events that may change the list of loads during during security analysis (disconnect / disable events). Have you taken this into account ? I will try to review the code with this in mind and see if there is a way to centralize more this logic).